daml/ContingentClaims/Lifecycle.daml

```diff
@@ -78,20 +78,26 @@ lifecycle' _ _ (OneF id) = do
   pure $ One id
 lifecycle' _ _ (GiveF (_, m)) = local negate m
 lifecycle' spot _ (WhenF obs (c, m)) = do
-  wasLifecycled <- lift . lift $ eval spot obs
-  if wasLifecycled then m $> Zero else pure $ When obs c
+  predicate <- lift . lift $ eval spot obs
+  if predicate then m $> Zero else pure $ When obs c
 lifecycle' spot _ (ScaleF obs (_, m)) = do
   k <- lift . lift $ eval spot obs
-  local (* k) m
+  Scale obs <$> local (* k) m
```

Comment:
I'm not certain this is write. I'll comment in the main thread about a test case.

Reply:
I think this should be
```
Scale (DecimalConst k) <$> local (* k) m
```